Functions/xASL_bids_BIDSifyASLJSON.m

```diff
@@ -173,13 +173,24 @@
 end
     
 %% 7. Check for Hadamard sequence
+
+% Hadamard Check
+if isfield(jsonOut,'TimeEncodedMatrix') && ~isempty(jsonOut,'TimeEncodedMatrix') || ... % Should be 4, 8 or 12
+    isfield(jsonOut,'HadamardMatrixType') % Natural or walsh
+    bHadamard = true; 
+else 
+    bHadamard = false; 
+end
+
+
 % FME sequence check
 if isfield(jsonIn,'SeriesDescription')
-	isHadamardFME = ~isempty(regexp(char(jsonIn.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+	bHadamardFME = ~isempty(regexp(char(jsonIn.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
 else
-	isHadamardFME = false;
+	bHadamardFME = false;
 end
-if isHadamardFME
+
+if bHadamardFME || bHadamard
```

Comment:
If `bHadamardFME` is true, `bHadamard` must be true right? So why do you say either or here, the question here should only be if `bHadamard` right?

Reply:
Yes, let's simplify this 👍

Reply:
bHadamard -> bTimeEncoded